src/kerberos/mod.rs

```diff
@@ -153,16 +156,43 @@ impl Kerberos {
 
     fn send(&self, data: &[u8]) -> Result<Vec<u8>> {
         if let Some((realm, kdc_url)) = self.get_kdc() {
-            return match kdc_url.scheme() {
-                "udp" | "tcp" => self.config.network_client.send(&kdc_url, data),
-                "http" | "https" => self.config.network_client.send_http(&kdc_url, data, Some(realm)),
-                scheme => {
-                    error!(?scheme, "Bad kdc url: invalid URL scheme");
-
-                    Err(Error::new(
-                        ErrorKind::InvalidParameter,
-                        "Invalid KDC server URL protocol scheme",
-                    ))
+            let protocol = NetworkProtocol::from_url_scheme(kdc_url.scheme()).ok_or_else(|| {
+                Error::new(
+                    ErrorKind::InvalidParameter,
+                    format!("Invalid protocol `{}` for KDC server", kdc_url.scheme()),
+                )
+            })?;
+
+            if !self.config.network_client.is_protocol_supported(protocol) {
+                return Err(Error::new(
+                    ErrorKind::InvalidParameter,
+                    format!(
+                        "Network protocol `{}` is not supported by `{}` network client. Supported protocols are: {:?}",
+                        kdc_url.scheme(),
+                        self.config.network_client.name(),
+                        self.config.network_client.supported_protocols(),
+                    ),
+                ));
+            }
+
+            return match protocol {
+                NetworkProtocol::Tcp | NetworkProtocol::Udp => {
+                    self.config.network_client.send(protocol, &kdc_url, data)
+                }
+                NetworkProtocol::Http | NetworkProtocol::Https => {
```

Comment:
Now `IronRDP` and other crates which depend on `sspi-rs` don't need to know about `KdcProxyMessage`, `picky`, etc.